tests/browser/test_tabhistory.py

```diff
@@ -0,0 +1,139 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2015 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for webelement.tabhistory."""
+
+from PyQt5.QtCore import QUrl, QPoint
+import pytest
+
+from qutebrowser.browser import tabhistory
+from qutebrowser.browser.tabhistory import TabHistoryItem as Item
+from qutebrowser.utils import qtutils
+
+
+class TestSerializeHistory:
+
+    """Tests for serialize()."""
+
+    ITEMS = [
+        Item(QUrl('https://www.heise.de/'), QUrl('http://www.heise.de/'),
+             'heise'),
+        Item(QUrl('http://example.com/%E2%80%A6'),
+             QUrl('http://example.com/%E2%80%A6'), 'percent', active=True),
+        Item(QUrl('http://example.com/?foo=bar'),
+             QUrl('http://original.url.example.com/'), 'arg',
+             user_data={'foo': 23, 'bar': 42}),
+        # From https://github.com/OtterBrowser/otter-browser/issues/709#issuecomment-74749471
+        Item(
+            QUrl('http://github.com/OtterBrowser/24/134/2344/otter-browser/'
+                 'issues/709/'),
+            QUrl('http://github.com/OtterBrowser/24/134/2344/otter-browser/'
+                 'issues/709/'),
+            'Page not found | github',
+            user_data={'zoom': 149, 'scroll-pos': QPoint(0, 0)}),
+        Item(
+            QUrl('https://mail.google.com/mail/u/0/#label/some+label/'
+                 '234lkjsd0932lkjf884jqwerdf4'),
+            QUrl('https://mail.google.com/mail/u/0/#label/some+label/'
+                 '234lkjsd0932lkjf884jqwerdf4'),
+            '"some label" - email@gmail.com - Gmail"',
+            user_data={'zoom': 120, 'scroll-pos': QPoint(0, 0)}),
+    ]
+
+    @pytest.fixture(autouse=True)
+    def setup(self, webpage):
```

Comment:
Problem with `setup_method` is that they can't receive fixtures as arguments, which is needed in several of the `setup` methods of classes I converted (and in this case, we use the `webpage` fixture). 

Those few that don't need any fixtures could instead declare `setup_method(self, method)` I suppose, but I rather use fixtures in all of them for consistency and because I believe fixtures are superior. :smile: